gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -497,6 +500,17 @@ def to_zarr(self, dataset: xr.Dataset, *args, **kwargs):
         **kwargs
             Keyword arguments to forward to `xr.Dataset.to_zarr`
         """
+        # Aggressively assert that the data is in contiguous time order. This is to protect against data corruption, especially during insert and append
+        # operations, but it should probably be replaced with a more sophisticated set of flags that let the user decide how to handle time data at their
+        # own risk.
+        #
+        # The expected delta is taken from the first two timestamps, but it could be improved by using Attributes.temporal_resolution if that function
+        # were edited to return a datetime64.
+        times = dataset[self.time_dim].values
+        expected_delta = times[1] - times[0]
+        if not self.are_times_contiguous(times, expected_delta):
+            raise ValueError("Dataset does not contain contiguous time data in " + dataset[self.time_dim])
```

Comment:
I think this exception will itself raise an exception when it tries to concatenate a string and a data array